[WFLY-16968]: WildFly High Availability Guide: Messaging write up

ehsavoie opened the pull request:
Adding some initial documentation on HA with Artemis. Based on Artemis documentation

Jira: https://issues.redhat.com/browse/WFLY-16968
      https://issues.redhat.com/browse/WFLY-20749

bstansberry commented on docs/src/main/asciidoc/_high-availability/Messaging.adoc:
```diff
@@ -8,4 +8,460 @@ ifdef::env-github[]
+The 2 backups with `group-name=bird` (`backup5` and `backup6`) will pair with primary servers `primary4` and `primary5`.
+====
+
+If `group-name` is not configured then the secondary will search for any primary that it can find the cluster.
```
s/find the cluster/find in the cluster/g

I think ^^^ is the intended meaning.

rhusar commented:
@ehsavoie This does looks like almost a verbatim copy from the  Artemis documentation. Is there any reason why not just link artemis documentation for things that are not WildFly-specific?

rhusar reviewed (changes requested):
Plus still missing shared storage recommendation for users.

bstansberry commented on docs/src/main/asciidoc/_high-availability/Messaging.adoc:
```diff
@@ -8,4 +8,448 @@ ifdef::env-github[]
+<replication-secondary allow-failback="true"/>
+----
+
+If the attribute `check-for-active-server` is set to `true` in the `primary` configuration then during start-up the primary server will first search the cluster for another active server using its nodeID.
```
What's a nodeID?

This is the only use of this string in the PR.

bstansberry commented on docs/src/main/asciidoc/_high-availability/Messaging.adoc:
```diff
@@ -8,4 +8,448 @@ ifdef::env-github[]
+After joining the cluster the backups with `group-name=fish` will search for primary servers with `group-name=fish` to pair with.
+Since there is one secondary too many, the `fish` will remain with one spare secondary.
+
+The 2 backups with `group-name=bird` (`backup5` and `backup6`) will pair with primary servers `primary4` and `primary5`.
```
This examples leaves me with lots of questions. I've not seen anything elsewhere in the docs that provides answers. If I've missed something we should link to it.

This describes, for each group, multiple primary/secondary pairs. Is there anywhere where we document the client semantics of sending and consuming messages in such a setup?

bstansberry commented on docs/src/main/asciidoc/_high-availability/Messaging.adoc:
```diff
@@ -8,4 +8,448 @@ ifdef::env-github[]
+
+If the attribute `check-for-active-server` is set to `true` in the `primary` configuration then during start-up the primary server will first search the cluster for another active server using its nodeID.
+If it finds one it will contact this server and try to "failback".
+Since this is a remote replication scenario the primary will have to synchronize its data with the secondary server running with its ID. Once they are in sync it will request the other server (which it assumes it is a secondary that has assumed its duties) to shutdown in order for it to take over.
```
Another, slightly different reference to an otherwise undescribed ID.

bstansberry commented on docs/src/main/asciidoc/_high-availability/Messaging.adoc:
```diff
@@ -8,4 +8,448 @@ ifdef::env-github[]
+An alternative to using primary/secondary groups is to configure _scaledown_.
+When configured for scale down a server can copy all its messages and transaction state to another active server.
+The advantage of this is that you don't need full backups to provide some form of HA, however there are disadvantages with this approach the first being that it only deals with a server being stopped and not a server crash.
+The caveat here is if you configure a secondary to scale down.
```
This is not a complete sentence.

ehsavoie commented on docs/src/main/asciidoc/_high-availability/Messaging.adoc:
```diff
@@ -8,4 +8,448 @@ ifdef::env-github[]
+
+failover-on-shutdown::
+The xref:wildscribe/subsystem/messaging-activemq/server/ha-policy/shared-store-primary/index.html[failover-on-shutdown] attribute describes whether the _graceful_ shutdown of this broker will cause the backup to activate.
+If `false` then the backup server will remain passive if this broker is shutdown gracefully (e.g. using kbd:[Ctrl+C]).
```
yes , it needs the experimental header

ehsavoie committed:
[WFLY-16968]: WildFly High Availability Guide: Messaging write up

Adding some initial documentation on HA with Artemis.
Based on Artemis documentation

Jira: https://issues.redhat.com/browse/WFLY-16968

Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

bstansberry reviewed (approved):
I still have things I'd like to see improved here, but this provides a lot of info about something that's not covered at all otherwise, and none of my concerns are about this saying something wrong; they are just about more/better explanation. So thanks and +1 for getting this into the WF 38 docs.
